### ISSUE TITLE
Take optional flag in whitespace rules into account

### ISSUE THREAD

**Issue by pmentz** [palantir/tslint#811]
Hi,

I have set

``` JSON
"typedef-whitespace": [
  true,
  {
    "parameter": "space"
  }
]
```

to enforce a space in front of the colon in method signatures. Unfortunately, what it seems to do is to force me to add a space after the parameter name, which only allows the following syntax for optional parameters:

``` Javascript
public myMethod(name ?:string):void { /* Looks like an optional type */ }
public myAlternateMethod(name ? :string):void { /* Looks like a ternary operator */ }
```

which is not quite, what I want to have.
For me, it would be very nice, if I could define this syntax:

``` Javascript
public myMethod(name? :string):void { /* empty*/ }
```

Am I overseeing a whitespace-rule for the optional flag?

**Comment by JKillian** [comment]
This seems like a bug to me. I can't imagine anyone actually wants a space between the name and the ? of parameters (or fields or anything similar). Unfortunately there isn't any workaround that I know of.

**Comment by b0r3as** [comment]
Shouldn't this be fixed with #958? Before the PR, the rule would take the end of the parameter name, advance one character and check that one for being a space (which is obviously wrong for the optional flag). The PR fixed this by explicitly searching for the colon and checking the character in front of it.

**Comment by adidahiya** [comment]
@b0r3as Good call, I think you're right. We should add some regression tests to verify the fix.

**Comment by th0r** [comment]
Another lint error that I faced after updating tslint from `5.8.0` to `5.9.1`:

![image](https://user-images.githubusercontent.com/302213/38558554-2b309bf4-3cd9-11e8-9225-1108cf568c93.png)

I use `?` to specify optional class property (and seems like it's a valid Typescript).

**Comment by liuxh0** [comment]
Same issue using tslint 5.11.0

**Comment by JoshuaKGoldberg** [comment]
Per #4534, this is an old issue with a formatting concern that has aged away.

**Comment by ataravati** [comment]
Why is this issue closed without being resolved?

**Comment by JoshuaKGoldberg** [comment]
Hi @ataravati! See #4534 (as linked in the comment just above yours). TSLint is deprecated and we're no longer accepting PRs for non-critical bugs such as this one.

**Comment by ataravati** [comment]
Thank you, @JoshuaKGoldberg!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
